packages/paste-libraries/react-textarea-autosize/src/index.tsx

```diff
@@ -0,0 +1,90 @@
+import type {SizingData} from './getSizingData';
+import {getSizingData} from './getSizingData';
+import {useComposedRef, useWindowResizeListener} from './hooks';
+import {noop} from './utils';
+
+type TextareaProps = React.TextareaHTMLAttributes<HTMLTextAreaElement>;
+
+type Style = Omit<NonNullable<TextareaProps['style']>, 'maxHeight' | 'minHeight'> & {
+  height?: number;
+};
+
+export type TextareaHeightChangeMeta = {
+  rowHeight: number;
+};
+export interface TextareaAutosizeProps extends Omit<TextareaProps, 'style'> {
+  maxRows?: number;
+  minRows?: number;
+  onHeightChange?: (height: number, meta: TextareaHeightChangeMeta) => void;
+  cacheMeasurements?: boolean;
+  style?: Style;
+}
+
+const TextareaAutosize: React.ForwardRefRenderFunction<HTMLTextAreaElement, TextareaAutosizeProps> = (
+  {cacheMeasurements, maxRows, minRows, onChange = noop, onHeightChange = noop, ...props},
+  userRef: React.Ref<HTMLTextAreaElement>
+) => {
+  if (process.env.NODE_ENV !== 'production' && props.style) {
+    if ('maxHeight' in props.style) {
+      throw new Error('Using `style.maxHeight` for <TextareaAutosize/> is not supported. Please use `maxRows`.');
+    }
+    if ('minHeight' in props.style) {
+      throw new Error('Using `style.minHeight` for <TextareaAutosize/> is not supported. Please use `minRows`.');
+    }
+  }
+  const isControlled = props.value !== undefined;
+  const libRef = React.useRef<HTMLTextAreaElement | null>(null);
+  const ref = useComposedRef(libRef, userRef);
+  const heightRef = React.useRef(0);
+  const measurementsCacheRef = React.useRef<SizingData>();
+
+  const resizeTextarea = (): void => {
+    const node = libRef.current!;
+    const nodeSizingData =
+      cacheMeasurements && measurementsCacheRef.current ? measurementsCacheRef.current : getSizingData(node);
+
+    if (!nodeSizingData) {
+      return;
+    }
+
+    measurementsCacheRef.current = nodeSizingData;
+
+    const [height, rowHeight] = calculateNodeHeight(
+      nodeSizingData,
+      node.value || node.placeholder || 'x',
+      minRows,
+      maxRows
+    );
+
+    if (heightRef.current !== height) {
+      heightRef.current = height;
+      node.style.setProperty('height', `${height}px`, 'important');
+      onHeightChange(height, {rowHeight});
+    }
+  };
+
+  const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>): void => {
+    // There isn't a user resize event on the textarea, so this is a low-cost way to check if the user manually resized it
+    // By default, when the user resizes a textarea, the new height is set in the style attribute
+    // The way the react-textarea-autosize works, it calculates the height and sets the style attribute with !important
+    // So, we can assume that if the height has important it is the autosize height and if it doesn't have important the user resized it themselves
+    const didUserResize = libRef.current?.style.getPropertyPriority('height') !== 'important';
```

Comment:
This is the change I made from the react-textarea-autosize library.

Reply:
todo: Could you please leave a comment about what the change was?

Reply:
I thought the comment above explains the change?

Reply:
counterpoint is, because you created our own version, you only really need to explain what _this_ does going forward. No need to explain what is does differently to a package we no longer use

Reply:
Ok - to clarify, the change is adding `didUserResize`. Because it is a hack, I added the comment

```
// There isn't a user resize event on the textarea, so this is a low-cost way to check if the user manually resized it
// By default, when the user resizes a textarea, the new height is set in the style attribute
// The way the react-textarea-autosize works, it calculates the height and sets the style attribute with !important
// So, we can assume that if the height has important it is the autosize height and if it doesn't have important the user resized it themselves
```

to explain why the hack works to distinguish between autosizing and user resizing.